### PR TITLE
Remove use of `uopz` extension in tests

### PR DESCRIPTION
**Pull request by benbowler**
## Summary

Addresses issue:

- #12332

## Relevant technical choices

- Added `permissions: contents: read` to `php-lint-tests.yml`, beyond the IB to fix the [zizmor CI failure](https://github.com/google/site-kit-wp/actions/runs/34207348179?pr=13521). The Actions scan only audits workflow files a PR touches, so editing this one surfaced four pre-existing `excessive-permissions` findings that blocked CI. Most other workflows still lack the block and will trip the same check when next edited.
- The refactored tests assert membership relative to the current environment rather than a literal `development` one, so both were renamed. Default production behaviour stays covered by `test_can_activate_on_post_5_5_version`.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [ ] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.
- [ ] Ensure there are no unexpected significant changes to file sizes.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: tests/phpunit/integration/Core/Tags/Guards/Tag_Environment_Type_GuardTest.php [hunk 1]
```diff
@@ -33,44 +38,43 @@ public function test_can_activate_on_older_versions() {
...
+		$current_env = wp_get_environment_type();
+
+		add_filter(
+			'googlesitekit_allowed_tag_environment_types',
+			function () use ( $current_env ) {
+				return array_values(
+					array_diff( array( 'local', 'development', 'staging', 'production' ), array( $current_env ) )
+				);
+			}
+		);
```
[the remark this answers is not in the document]
Comment (benbowler) [reply tests/phpunit/integration/Core/Tags/Guards/Tag_Environment_Type_GuardTest.php:56]: Leaving this as is. An empty array makes the `in_array()` check trivially false, so the test would still pass against a guard that never checks membership at all, for example one that only returned whether the allowed list was non-empty. Passing a non-empty list that excludes the current environment is what actually pins that behaviour down.

This is also the approach the IB specified. An empty allowed list is a reasonable edge case, but it would be an extra test rather than a replacement for this one.

**Review by zutigrm — APPROVED** [review]
LGTM

**Review by eugene-manuilov — APPROVED** [review]
